### PR TITLE
refactor: separate Normalizer and Ingester

### PR DESCRIPTION
**Pull request by Cluas**
This PR addresses the problem described in the issue by separating the two components.
#4169

### CODE REVIEW

**Comment by Cluas** [comment]
By making this modification, we can use a different storage for the profile store, which will become easy to do. We just need to implement the Ingester ourselves, without the need for coupling with the Normalizer.

File: pkg/parca/parca.go [hunk 1]
```diff
@@ -108,7 +111,8 @@ type Flags struct {
 	Debuginfo  FlagsDebuginfo  `embed:"" prefix:"debuginfo-"`
 	Debuginfod FlagsDebuginfod `embed:"" prefix:"debuginfod-"`
 
-	Metastore string `default:"badger" help:"Which metastore implementation to use" enum:"badger"`
+	Metastore    string `default:"badger" help:"Which metastore implementation to use" enum:"badger"`
+	Profilestore string `default:"frostdb" help:"Which profilestore implementation to use" enum:"frostdb"`
```
Comment (brancz) [inline pkg/parca/parca.go:115]: I would say let's skip a flag until we actually decide on something as part of the Parca project itself.

File: go.mod [hunk 2]
```diff
@@ -59,6 +59,12 @@ require (
 	gopkg.in/yaml.v3 v3.0.1
 )
 
+require (
```
Comment (brancz) [inline go.mod:62]: let's merge this with the below block

**Review by brancz — APPROVED** [review]
Nice! lgtm let's just remove the flag until we actually have other options.

File: go.mod (hunk 2 above)
Comment (Cluas) [reply go.mod]: Done

**Review by thorfour — APPROVED** [review]
LGTM!

**Pull request merged**